Content.Server/Atmos/EntitySystems/DeltaPressureSystem.cs

```diff
@@ -0,0 +1,283 @@
+using Content.Server.Atmos.Components;
+using Content.Shared.Damage;
+using Content.Shared.Examine;
+using JetBrains.Annotations;
+
+namespace Content.Server.Atmos.EntitySystems;
+
+/// <summary>
+/// <para>System that handles <see cref="DeltaPressureComponent"/>.</para>
+///
+/// <para>Entities with a <see cref="DeltaPressureComponent"/> will take damage per atmostick
+/// depending on the pressure they experience.</para>
+///
+/// <para>DeltaPressure logic is mostly handled in a partial class in Atmospherics.
+/// This system handles the adding and removing of entities to a processing list,
+/// as well as any field changes via the API.</para>
+/// </summary>
+public sealed class DeltaPressureSystem : EntitySystem
+{
+    [Dependency] private readonly AtmosphereSystem _atmosphereSystem = default!;
+    [Dependency] private readonly DamageableSystem _damage = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<DeltaPressureComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<DeltaPressureComponent, ComponentShutdown>(OnComponentShutdown);
+        SubscribeLocalEvent<DeltaPressureComponent, ExaminedEvent>(OnExamined);
+
+        SubscribeLocalEvent<DeltaPressureComponent, GridUidChangedEvent>(OnGridChanged);
+    }
+
+    private void OnComponentInit(Entity<DeltaPressureComponent> ent, ref ComponentInit args)
+    {
+        if (ent.Comp.AutoJoin)
+        {
+            TryAddToList(ent);
+        }
+    }
+
+    private void OnComponentShutdown(Entity<DeltaPressureComponent> ent, ref ComponentShutdown args)
+    {
+        TryRemoveFromList(ent);
+    }
+
+    private void OnExamined(Entity<DeltaPressureComponent> ent, ref ExaminedEvent args)
+    {
+        if (ent.Comp.IsTakingDamage)
+            args.PushMarkup(Loc.GetString("window-taking-damage"));
+    }
+
+    private void OnGridChanged(Entity<DeltaPressureComponent> ent, ref GridUidChangedEvent args)
+    {
+        if (args.OldGrid != null)
+        {
+            TryRemoveFromList(ent, args.OldGrid.Value);
+        }
+
+        if (args.NewGrid != null)
+        {
+            TryAddToList(ent, args.NewGrid.Value);
+        }
+    }
+
+    /// <summary>
+    /// Attempts to add an entity to the list of entities to do comparison and damage work on in AtmosphereSystem.
+    /// </summary>
+    /// <param name="ent">The entity to add.</param>
+    /// <returns>True if the device was added, false if the device could not be added
+    /// or was already in the processing list.</returns>
+    [PublicAPI]
+    public bool TryAddToList(Entity<DeltaPressureComponent> ent)
```

Comment:
This isn't a great method name. "List" is just super vague. Thinking over it some more, maybe "TryQueueProcessing" would be better verbage?